bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java

```diff
@@ -0,0 +1,267 @@
+            } catch (IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+        } else {
+            modified(properties);
+        }
+    }
+
+    @Modified
+    public void modified(Map<String, Object> properties) {
+        this.config = new Configuration(properties).as(HueEmulationConfig.class);
+
+        if (config.discoveryHttpPort == 0) {
+            config.discoveryHttpPort = Integer.getInteger("org.osgi.service.http.port", 8080);
+        }
+
+        String discoveryIp = config.discoveryIp;
+        if (discoveryIp == null) {
+            discoveryIp = networkAddressService.getPrimaryIpv4HostAddress();
+        }
+
+        if (discoveryIp == null) {
+            logger.warn("No primary IP address configured. Discovery disabled!");
+            return;
+        }
+
+        // Get and apply configurations
+        ds.config.linkbutton = config.pairingEnabled;
+        ds.config.createNewUserOnEveryEndpoint = config.createNewUserOnEveryEndpoint;
+        ds.config.networkopenduration = config.pairingTimeout;
+        ds.config.devicename = config.devicename;
+
+        ds.config.uuid = config.uuid;
+        ds.config.bridgeid = config.uuid.replace("-", "").toUpperCase();
+        if (ds.config.bridgeid.length() > 12) {
+            ds.config.bridgeid = ds.config.bridgeid.substring(0, 12);
+        }
+
+        pairingTimeout.check(configAdmin, ds);
+
+        try {
+            address = InetAddress.getByName(discoveryIp);
+            if (eventAdmin != null) {
+                eventAdmin.postEvent(new Event(UpnpServer.TOPIC_ADDRESS_CHANGED, Collections.emptyMap()));
+            }
+        } catch (UnknownHostException e) {
+            logger.warn("No primary IP address configured. Discovery disabled!", e);
+            address = InetAddress.getLoopbackAddress();
+        }
+        ds.config.mac = NetworkUtils.getMAC(address);
+        ds.config.ipaddress = address.getHostAddress();
+    }
+
+    protected void determineHighestAssignedHueID() {
+        for (Metadata metadata : metadataRegistry.getAll()) {
+            if (!metadata.getUID().getNamespace().equals(METAKEY)) {
+                continue;
+            }
+            try {
+                int hueId = Integer.parseInt(metadata.getValue());
+                if (hueId > highestAssignedHueID) {
+                    highestAssignedHueID = hueId;
+                }
+            } catch (NumberFormatException e) {
+                logger.warn("A non numeric hue ID '{}' was assigned. Ignoring!", metadata.getValue());
+            }
+        }
+    }
+
+    /**
+     * Although hue IDs are strings, a lot of implementations out there assume them to be numbers. Therefore
+     * we map each item to a number and store that in the meta data provider.
+     *
+     * @param item The item to map
+     * @return A stringified integer number
+     */
+    public String mapItemUIDtoHueID(@Nullable Item item) {
```

Comment:
Would it make more sense to use a different annotation, if you want to guard against null?